lib/matplotlib/axes/_axes.py

```diff
@@ -6091,30 +6091,31 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                              "Please only use 'density', since 'normed'"
                              "will be deprecated.")
 
-        # process the unit information
-        self._process_unit_info(xdata=x, kwargs=kwargs)
-        x = self.convert_xunits(x)
-        if bin_range is not None:
-            bin_range = self.convert_xunits(bin_range)
-
-        # Check whether bins or range are given explicitly.
-        binsgiven = (cbook.iterable(bins) or bin_range is not None)
-
         # basic input validation
         input_empty = np.size(x) == 0
-
         # Massage 'x' for processing.
         if input_empty:
-            x = np.array([[]])
+            x = [np.array([[]])]
```

Comment:
Too many square brackets--I think this should be a list with one empty 1-D array, not with an empty 2-D array.